runtime/expr/function/ip.go

```diff
@@ -30,15 +30,21 @@ func (n *NetworkOf) Call(ctx zed.Allocator, args []zed.Value) *zed.Value {
 		}
 	} else {
 		// two args
-		id := args[1].Type.ID()
 		body := args[1].Bytes
-		if id == zed.IDNet {
+		switch id := args[1].Type.ID(); {
+		case id == zed.IDNet:
 			cidrMask := zed.DecodeNet(body)
 			if !bytes.Equal(cidrMask.IP, cidrMask.Mask) {
 				return newErrorf(n.zctx, ctx, "network_of: network arg not a cidr mask")
 			}
 			mask = cidrMask.Mask
-		} else if zed.IsInteger(id) {
+		case id == zed.IDIP:
+			ip := zed.DecodeIP(body)
+			mask = net.IPMask(ip.AsSlice())
+			if ones, bits := mask.Size(); ones == 0 && bits == 0 {
+				return newErrorf(n.zctx, ctx, "network_of: mask %q is not in canonical form", ip.String())
```

Comment:
Nit: No real need to quote this, and I think the embedded quotes it produces make the error string less readable.
```suggestion
				return newErrorf(n.zctx, ctx, "network_of: mask %s is not in canonical form", ip.String())
```